Marlin/src/pins/stm32f1/pins_CREALITY_V24S1_301.h

```diff
@@ -44,7 +44,12 @@
 //
 // Limit Switches
 //
-#define Z_STOP_PIN                          PC14
+#ifdef BLTOUCH
+  #define Z_STOP_PIN                        PC14
+  #define Z_STOP_PIN_NADD                   PA15  //Added z-axis limit switch  rock_20210816
+#else
+  #define Z_STOP_PIN                        PA15
+#endif
```

Comment:
Since this is selected according to `Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN` we should not apply conditional pin exchanges in pins files, especially for straightforward peripherals.

Reply:
That is coming from official sources from Creality, I tested that compiling a Manual Mesh version that uses the spare Z end stop switch connected to:
![image](https://user-images.githubusercontent.com/2745567/157812617-576b2434-f08e-4f85-9c19-3b05c1f08270.png)

Reply:
If the BLTouch is connected to a pin that is not the Z endstop pin, the configuration should have `Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN` disabled. The probe will then use `PC14` while leaving `Z_STOP_PIN` defined as `PA15` (in case someone wants to use it for another purpose).

There should be no need to fudge the `Z_STOP_PIN`, which would affect all configurations regardless of whether `Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN` is enabled. If `Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN` should never be enabled with this board, that should be told to the user with an `#error`.

Reply:
I agree 'Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN' must be disabled. Did you see the official Creality repo for the S1?

Reply:
No, I have not looked at it.

So if `Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN`must be disabled with this board, then an `#error` is the proper way to enforce it.